pkg/asset/store/store.go

```diff
@@ -282,6 +304,11 @@ func (s *storeImpl) load(a asset.Asset, indent string) (*assetState, error) {
 			if err := s.loadAssetFromState(stateFileAsset); err != nil {
 				return nil, errors.Wrapf(err, "failed to load asset %q from state file", a.Name())
 			}
+
+			if strings.Compare(reflect.TypeOf(stateFileAsset).String(), "*tls.KubeletCSRSignerCertKey") == 0 {
+				checkExpiredSigner(stateFileAsset.(*tls.KubeletCSRSignerCertKey))
+			}
```

Comment:
This change un acceptably  ties the implementation of the generic store and the assets that are created using it.

The store understands how to create Assets using the Generate and Dependencies defined on them, it doesn't really care what that asset is as long as it satisfies the interface. All the actual specific code lives in the Asset itself. Here you have added a conditional to the store based on one of the asset names creating a coupling that breaks that boundary/abstraction.

Reply:
Where would be a better place to make this check?  I'd like to help if I can.  Thanks!

Reply:
Seems like we need a `Load` style hook for ingesting from state, similar to the one we already have for ingesting from disk.